app/scripts/lib/createTracingMiddleware.ts

```diff
@@ -1,9 +1,31 @@
 // Request and repsones are currently untyped.
 /* eslint-disable @typescript-eslint/no-explicit-any */
 
-import { MESSAGE_TYPE } from '../../../shared/constants/app';
+import {
+  MESSAGE_TYPE,
+  TRACE_ENABLED_SIGN_METHODS,
+} from '../../../shared/constants/app';
 import { trace, TraceName } from '../../../shared/lib/trace';
 
+async function handleTracing(
+  req: any,
+  id: any,
+  traceName: TraceName,
+  tags?: Record<string, string>,
+) {
+  req.traceContext = await trace({
+    name: traceName,
```

Comment:
Rather than hiding this in another function, would it be more readable to include this directly within `tracingMiddleware` but use a map of method to trace name for the main trace?